internal/services/redhatopenshift/redhat_openshift_cluster_resource.go

```diff
@@ -135,6 +136,19 @@ func (r RedHatOpenShiftCluster) Arguments() map[string]*pluginsdk.Schema {
 						Sensitive:    true,
 						ValidateFunc: validation.StringIsNotEmpty,
 					},
+					"resource_group_name": {
```

Comment:
The name of this is quite confusing and clashes with the existing `resource_group_id`. Please give a more descriptive name for this.

Reply:
`resource_group_name` is to set the resource group which will be created by Azure RedHat OpenShift RP to deploy dependency resources. I have confirmed with service team they won't support RG in another subscription so I only expose it as `resource_group_name`. And before we expose this `resource_group_name` property, the resource group name is default to `aro-{domain}` as portal does, where `domain` is another configurable property. 

The `resource_group_id` is a computed property to display the above resource group ID. 

Do we need change this to `target_resource_group_name` or `dependency_resource_group_name`? Or can we keep it similar format to `resource_group_id`?

Reply:
> Separate question, can ARO only deploy it's resources to a resource group that it creates? Or can it deploy it's resources to an already existing one?

From service team, it must be a resource group that it creates. In other word, the resource group name sent in this field must be a non-existing one.